--- gcc-2.95.2/config.sub.chmou	Thu Jun 29 15:32:27 2000
+++ gcc-2.95.2/config.sub	Thu Jun 29 15:47:52 2000
@@ -1252,4 +1252,8 @@
 		;;
 esac
 
+if [ -f /etc/debian_version ]; then
+ basic_machine=`echo $basic_machine | sed -e 's/-pc/-mdk/' -e 's/-unknown/-mdk/'`
+fi
+
 echo $basic_machine$os
--- gcc-2.95.2/config.guess.chmou	Thu Jun 29 15:32:27 2000
+++ gcc-2.95.2/config.guess	Thu Jun 29 15:48:22 2000
@@ -819,9 +819,11 @@
 	  case "${UNAME_MACHINE}" in
 	  i?86)
 	    VENDOR=pc;
+	    if [ -f /etc/mandrake-release ];then VENDOR=mdk;fi;
 	    ;;
 	  *)
 	    VENDOR=unknown;
+	    if [ -f /etc/mandrake-release ];then VENDOR=mdk;fi;
 	    ;;
 	  esac
 	  # Determine whether the default compiler is a.out or elf
