--- libgcj/Makefile.in.jj	Tue Aug 10 16:01:37 1999
+++ libgcj/Makefile.in	Fri Mar  1 17:47:58 2002
@@ -777,7 +777,6 @@ ALL_TARGET_MODULES = \
 	all-target-libstub \
 	all-target-libffi \
 	all-target-libjava \
-	all-target-zlib \
 	all-target-boehm-gc \
 	all-target-qthreads \
 	all-target-bsp \
@@ -800,7 +799,6 @@ CONFIGURE_TARGET_MODULES = \
 	configure-target-libstub \
 	configure-target-libffi \
 	configure-target-libjava \
-	configure-target-zlib \
 	configure-target-boehm-gc \
 	configure-target-qthreads \
 	configure-target-bsp \
@@ -817,7 +815,6 @@ CHECK_TARGET_MODULES = \
 	check-target-libiberty \
 	check-target-libffi \
 	check-target-libjava \
-	check-target-zlib \
 	check-target-boehm-gc \
 	check-target-qthreads \
 	check-target-gperf
@@ -835,7 +832,6 @@ INSTALL_TARGET_MODULES = \
 	install-target-libiberty \
 	install-target-bsp \
 	install-target-libjava \
-	install-target-zlib \
 	install-target-boehm-gc \
 	install-target-qthreads \
 	install-target-gperf
@@ -920,7 +916,6 @@ CLEAN_TARGET_MODULES = \
 	clean-target-libstub \
 	clean-target-libffi \
 	clean-target-libjava \
-	clean-target-zlib \
 	clean-target-boehm-gc \
 	clean-target-qthreads \
 	clean-target-bsp \
@@ -1614,9 +1609,9 @@ check-target-libio: all-target-libstdc++
 all-libgui: all-tcl all-tk all-tcl8.1 all-tk8.1 all-itcl
 all-libiberty:
 configure-target-libffi: $(ALL_GCC) 
-configure-target-libjava: $(ALL_GCC) configure-target-zlib configure-target-boehm-gc configure-target-qthreads configure-target-newlib configure-target-libffi
+configure-target-libjava: $(ALL_GCC) configure-target-boehm-gc configure-target-qthreads configure-target-newlib configure-target-libffi
 all-target-libffi: configure-target-libffi
-all-target-libjava: configure-target-libjava all-gcc all-zip all-target-newlib all-target-zlib all-target-boehm-gc all-target-qthreads all-target-libffi
+all-target-libjava: configure-target-libjava all-gcc all-zip all-target-newlib all-target-boehm-gc all-target-qthreads all-target-libffi
 configure-target-librx: $(ALL_GCC) configure-target-newlib
 all-target-librx: configure-target-librx
 configure-target-libstdc++: $(ALL_GCC)
@@ -1660,8 +1655,6 @@ all-target-winsup: all-target-newlib all
 configure-target-winsup: configure-target-newlib
 all-uudecode: all-libiberty
 all-zip:
-configure-target-zlib: $(ALL_GCC)
-all-target-zlib: configure-target-zlib
 configure-target-libiberty: $(ALL_GCC)
 all-target-libiberty: configure-target-libiberty all-gcc all-ld all-target-newlib
 all-target: $(ALL_TARGET_MODULES)
